### PR TITLE
No more ninja

### PR DESCRIPTION
**Pull request by majestrate**
remove use of ninja from CI so that static builds can use the makefile job server and thus go faster.

### CODE REVIEW

File: contrib/android.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+set -e
+set -x
+test x$NDK = x  && exit 1
+root="$(readlink -f $(dirname $0)/../)"
+out=$root/lokinet-android-$(git describe)
+mkdir -p $out
+mkdir -p $root/build-android
+cd $root/build-android
+for abi in armeabi-v7a arm64-v8a x86 x86_64 ; do
```
Comment (tewinget) [inline contrib/android.sh]: prefer to define the ABI list in a separate line above so it's clear where one should go to edit that list if ever we need to.

Comment (majestrate) [reply contrib/android.sh]: fixed in https://github.com/oxen-io/loki-network/pull/1673/commits/b7485b4b25bfab4033045bfaaa0f6a242a69b814

File: .drone.jsonnet [hunk 2]
```diff
@@ -49,20 +49,20 @@ local debian_pipeline(name, image,
...
+                'eatmydata ' + apt_get_quiet + ' install -y gdb cmake git pkg-config ccache ' + deps,
                 'mkdir build',
                 'cd build',
-                'cmake .. -G Ninja -DWITH_SETCAP=OFF -DCMAKE_CXX_FLAGS=-fdiagnostics-color=always -DCMAKE_BUILD_TYPE='+build_type+' ' +
+                'cmake .. -DWITH_SETCAP=OFF -DCMAKE_CXX_FLAGS=-fdiagnostics-color=always -DCMAKE_BUILD_TYPE='+build_type+' ' +
                     (if werror then '-DWARNINGS_AS_ERRORS=ON ' else '') +
                     '-DWITH_LTO=' + (if lto then 'ON ' else 'OFF ') +
                 cmake_extra,
-                'ninja -j' + jobs + ' -v',
+                'make -j' + jobs,
```
Comment (jagerman) [inline .drone.jsonnet]:
```suggestion
                'make -j' + jobs + ' VERBOSE=1`,
```

File: llarp/util/fs.hpp [hunk 3]
```diff
@@ -57,10 +57,10 @@ namespace llarp
     OpenFileStream(fs::path pathname, std::ios::openmode mode)
     {
       if (EnsurePrivateFile(pathname))
-        return {};
+        return std::nullopt;
 
       std::string f = pathname.string();
-      return T{pathname, mode};
+      return T{f, mode};
```
Comment (jagerman) [inline llarp/util/fs.hpp]: Will `return std::make_optional<T>(pathname, mode);` work here?

File: .drone.jsonnet [hunk 4]
```diff
@@ -113,17 +109,10 @@ local windows_cross_pipeline(name, image,
...
                 'update-alternatives --set x86_64-w64-mingw32-g++ /usr/bin/x86_64-w64-mingw32-g++-posix',
-                'mkdir build',
-                'cd build',
-                'cmake .. -G Ninja -DCMAKE_EXE_LINKER_FLAGS=-fstack-protector -DCMAKE_CXX_FLAGS=-fdiagnostics-color=always -DCMAKE_TOOLCHAIN_FILE=../contrib/cross/mingw'+toolchain+'.cmake -DCMAKE_BUILD_TYPE='+build_type+' ' +
-                    (if werror then '-DWARNINGS_AS_ERRORS=ON ' else '') +
-                    (if lto then '' else '-DWITH_LTO=OFF ') +
-                    "-DBUILD_STATIC_DEPS=ON -DDOWNLOAD_SODIUM=ON -DBUILD_PACKAGE=ON -DBUILD_SHARED_LIBS=OFF -DBUILD_TESTING=OFF -DNATIVE_BUILD=OFF -DSTATIC_LINK=ON" +
-                cmake_extra,
-                'ninja -j' + jobs + ' -v package',
+                'VERBOSE=1 JOBS=' + jobs + ' ./contrib/windows.sh'
```
Comment (jagerman) [inline .drone.jsonnet:115]: Missing the `make`?

Comment (majestrate) [reply .drone.jsonnet:115]: it's calling`./contrib/windows.sh` which sets up cmake and does `make package`, i changed it to this for DRY related reasons.

File: llarp/util/fs.hpp (hunk 3 above)
Comment (majestrate) [reply llarp/util/fs.hpp]: yes it does

**Review by jagerman — APPROVED** [review]

**Pull request merged**